community/kernel/src/main/java/org/neo4j/kernel/impl/proc/ReflectiveProcedureCompiler.java

```diff
@@ -376,19 +392,39 @@ private ReflectiveUserAggregationFunction compileAggregationFunction( Class<?> d
         MethodHandle creator = lookup.unreflect( method );
         MethodHandle updateMethod = lookup.unreflect( update );
         MethodHandle resultMethod = lookup.unreflect( result );
-        List<FieldInjections.FieldSetter> setters = safeFieldInjections.setters( definition );
 
         Optional<String> description = description( method );
         UserAggregationFunction function = method.getAnnotation( UserAggregationFunction.class );
 
         Optional<String> deprecated = deprecated( method, function::deprecatedBy,
                 "Use of @UserAggregationFunction(deprecatedBy) without @Deprecated in " + funcName );
 
+        List<FieldInjections.FieldSetter> setters = allFieldInjections.setters( definition );
+        if ( !config.fullAccessFor( funcName.toString() ) )
+        {
+            try
+            {
+                setters = safeFieldInjections.setters( definition );
+            }
+            catch ( ComponentInjectionException e )
+            {
+                description = Optional.of( funcName.toString() +
+                        " is not available due to not having unrestricted access rights, check configuration." );
```

Comment:
how about avoiding double-negation here, and go for `is not available due to having restricted access rights, check configuration.`